adaptor/objectives/MLM.py

```diff
@@ -60,10 +60,13 @@ def _get_inputs_iterator(self, split: str) -> Iterable[Union[BatchEncoding, Dict
         :param split: Data split. `train` or `eval`.
         :return:
         """
-        if self.texts is not None:
-            texts_iter = iter(self.texts)
+        texts = self.texts if split == 'train' else self.val_texts
```

Comment:
Can be replaced with `texts = self._per_split_iterator_single(split)` -- this is clearly a bug, but I believe the functionality is already implemented in the inherited `UnsupervisedObjective`.

Reply:
Fixed in 9bd379f.